src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -674,6 +696,29 @@ protected int VariableSlot(Symbol symbol, int containingSlot = 0)
         protected int GetOrCreateSlot(Symbol symbol, int containingSlot = 0)
         {
             if (symbol is RangeVariableSymbol) return -1;
+
+            var fieldSymbol = symbol as TupleFieldSymbol;
+            if ((object)fieldSymbol != null)
+            {
+                TypeSymbol containingType = ((TupleTypeSymbol)symbol.ContainingType).UnderlyingNamedType;
+
+                // for tuple fields the varible indentifier represents the underlying field
+                symbol = fieldSymbol.TupleUnderlyingField;
+
+                // descend through Rest fields
+                // force corresponding slots if do not exist
+                while (containingType != symbol.ContainingType)
+                {
+                    var restField = containingType.GetMembers(TupleTypeSymbol.RestFieldName).FirstOrDefault() as FieldSymbol;
+                    if (restField == null)
+                    {
+                        return -1;
+                    }
+                    containingSlot = GetOrCreateSlot(restField, containingSlot);
+                    containingType = restField.Type.TupleUnderlyingTypeOrSelf();
+                }
+            }
```

Comment:
Would it be worth extracting a helper method for use here and in `VariableSlot` above: `int GetTupleFieldSlot(TupleFieldSymbol field, int containingSlot, bool createIfMissing)`?